### PR TITLE
fix(Scripts/TheBlackTemple): resetting behaviour in p1 for Shade of Akama

### PR DESCRIPTION
**Pull request by elthehablo**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20067
- Closes https://github.com/chromiecraft/chromiecraft/issues/7385#issue-2554017193

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Rorschach91** [comment]
PR tested, it works.
It can be reviewed.

**Comment by elthehablo** [comment]
> PR tested, it works.
> It can be reviewed.

Thank you for testing ❤️ I believe the factions are wrong though, so I plan on changing that today still

**Comment by elthehablo** [comment]
Right now the spawned mobs don't attack akama anymore

**Comment by Rorschach91** [comment]
Let me know when I can test it again.

**Comment by elthehablo** [comment]
> Let me know when I can test it again.

Ty. Now!

**Comment by elthehablo** [comment]
I'm just in a battle with the codestyle checks

**Comment by elthehablo** [comment]
Okay sorry. Should be good to test now

**Comment by Rorschach91** [comment]
I'll test it this afternoon. Thanks for the warning.

**Comment by Rorschach91** [comment]
Ehm....I don't think it is normal: [Youtube](https://youtu.be/5Zfs865UFR8)

**Comment by elthehablo** [comment]
Interesting...

**Comment by elthehablo** [comment]
@heyitsbench did some research. And all mobs should despawn on wipe other than the defenders. In the last commit the behaviour should be corrected. So feel free to test again :)

**Comment by Rorschach91** [comment]
Roger. Give me a minute.

**Comment by heyitsbench** [comment]
> if raid wipes and there are no defenders, Akama stays in that way whitout noone that can kill it.

Forgive me if I misunderstand, but afaik Defenders (and other Ashtongue creatures) should continue to spawn until Shade of Akama is fully unleashed and attacking Akama directly. If that's the case, then there will never be a softlock situation like this, as Shade of Akama will eventually kill Akama once attacking him, or Defenders will continue to spawn and attack Akama.

**Comment by Rorschach91** [comment]
> > if raid wipes and there are no defenders, Akama stays in that way whitout noone that can kill it.
> 
> Forgive me if I misunderstand, but afaik Defenders (and other Ashtongue creatures) should continue to spawn until Shade of Akama is fully unleashed and attacking Akama directly. If that's the case, then there will never be a softlock situation like this, as Shade of Akama will eventually kill Akama once attacking him, or Defenders will continue to spawn and attack Akama.

With this fix, mobs stop to spawn if raid wipes.

**Comment by Rorschach91** [comment]
> > if raid wipes and there are no defenders, Akama stays in that way whitout noone that can kill it.
> 
> Forgive me if I misunderstand, but afaik Defenders (and other Ashtongue creatures) should continue to spawn until Shade of Akama is fully unleashed and attacking Akama directly. If that's the case, then there will never be a softlock situation like this, as Shade of Akama will eventually kill Akama once attacking him, or Defenders will continue to spawn and attack Akama.

With this fix, if all raid is dead, all spawned mobs disappear except the defenders that attack Akama. 
This is my fear. If all mobs despawn and there are no defenders alive, noone will kill Akama, so the raid will be bugged.

**Comment by Rorschach91** [comment]
> @heyitsbench did some research. And all mobs should despawn on wipe other than the defenders. In the last commit the behaviour should be corrected. So feel free to test again :)

If you are sure about this, at least the Defenders should continue to appear until Akama dies.

**Comment by heyitsbench** [comment]
> I forgot to say that if all raid members die mobs stop to spawn.

This is not how it should work. Mobs should continue to spawn until (I believe) Akama is engaged by the Shade.

File: src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp [hunk 1]
```diff
@@ -232,7 +233,7 @@ struct npc_akama_shade : public ScriptedAI
             {
             case POINT_ENGAGE:
                 me->SetHomePosition(me->GetPosition());
-                me->SetFaction(FACTION_ENGAGE);
+                me->SetFaction(FACTION_MONSTER_SPAR_BUDDY);
```
Comment (heyitsbench) [inline src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp:236]: For the record, Akama is supposed to change his faction (twice), not the Shade.

Comment (elthehablo) [reply src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp:236]: This is akama, and not the shade. The naming is confusing

**Comment by elthehablo** [comment]
> > I forgot to say that if all raid members die mobs stop to spawn.
> 
> This is not how it should work. Mobs should continue to spawn until (I believe) Akama is engaged by the Shade.

Good to know. Will add that then. Thanks.

**Comment by elthehablo** [comment]
> > I forgot to say that if all raid members die mobs stop to spawn.
> 
> This is not how it should work. Mobs should continue to spawn until (I believe) Akama is engaged by the Shade.

So I take this as: if all players are dead and akama is still channeling, only defenders should still be spawning to go and attack akama

**Comment by heyitsbench** [comment]
Tested on retail PTR. If all players are dead while Akama is channeling, *all* of the usual mobs will continue to spawn. The mobs that usually would despawn with no combat also continue to spawn, but they do not despawn, they stand at their spawn point without moving. Defenders continue to run to Akama and beat him up, and the Sorcerers continue to run to the Shade and channel upon him.

**Comment by elthehablo** [comment]
But I don't see that getting fixed without weird workarounds, so it is good enough for me tbh

**Comment by heyitsbench** [comment]
I say that's a definite improvement, maybe best to open an issue just to remember the proper behavior and call it good.

**Comment by elthehablo** [comment]
> I say that's a definite improvement, maybe best to open an issue just to remember the proper behavior and call it good.

Yes that is fine

**Comment by elthehablo** [comment]
Would you care to do that? You have the deets :)

**Pull request merged**